File: hsqldb/jruby/create/hsqldb_create.rb
#! /usr/bin/jruby
# -*- encoding: utf-8 -*-
#
#	hsqldb_create.rb
#
#					Jan/29/2015
# ---------------------------------------------------------------------
include Java
import java.sql.DriverManager
import java.lang.System
#
#
load '/var/www/data_base/common/jruby_common/jruby_sql_manipulate.rb'
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
# ---------------------------------------------------------------------
def prepare_data_proc ()
	dict_aa={}
	dict_aa=dict_append_proc(dict_aa,"t4671","鹿児島",76132,"2007-9-24");
	dict_aa=dict_append_proc(dict_aa,"t4672","指宿",94258,"2007-8-12");
	dict_aa=dict_append_proc(dict_aa,"t4673","志布志",31658,"2007-5-11");
	dict_aa=dict_append_proc(dict_aa,"t4674","川内",57234,"2007-9-25");
	dict_aa=dict_append_proc(dict_aa,"t4675","鹿屋",12976,"2007-10-16");
	dict_aa=dict_append_proc(dict_aa,"t4676","枕崎",31547,"2007-11-2");
	dict_aa=dict_append_proc(dict_aa,"t4677","出水",69831,"2007-2-6");
	dict_aa=dict_append_proc(dict_aa,"t4678","阿久根",89432,"2007-5-29");
	dict_aa=dict_append_proc(dict_aa,"t4679","霧島",52816,"2007-10-17");

	return dict_aa
end
# ---------------------------------------------------------------------
puts	"*** 開始 ***"
#
str_connect = "jdbc:hsqldb:file:/var/tmp/hsqldb/cities;shutdown=true"
conn = DriverManager.getConnection(str_connect,"SA", "")
#
dict_aa=prepare_data_proc()
#
sql_drop_table_proc(conn)
sql_create_table_proc(conn)
#
dict_aa.each {|key,value|
	sql_insert_record_proc(conn,key,value['name'], \
		value['population'],value['date_mod'])
	}
#
sql_read_proc(conn)
#
conn.close()
puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------

File: hsqldb/clojure/create/hsqldb_create.clj
; -----------------------------------------------------------------
;
;	hsqldb_create.clj
;
;					Jul/17/2014
;
; -----------------------------------------------------------------
(import '(java.util Date))
(import '(java.text SimpleDateFormat))
(use 'clojure.java.jdbc)
; -----------------------------------------------------------------
(defn insert_proc []
    (insert-values :cities
        [:id :name :population :date_mod] ["t4671" "鹿児島" 21736 "1971-2-21"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4672" "指宿" 47328 "1971-9-8"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4673" "志布志" 72391 "1971-3-15"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4674" "川内" 42869 "1971-11-21"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4675" "鹿屋" 36158 "1971-10-7"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4676" "枕崎" 42917 "1971-7-12"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4677" "出水" 82534 "1971-2-24"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4678" "阿久根" 91487 "1971-5-21"])
    (insert-values :cities
        [:id :name :population :date_mod] ["t4679" "霧島" 46257 "1971-9-17"])
    )
; -----------------------------------------------------------------
(println" "*** 開始 ***")

(let [db-host "file"
      db-name "/var/tmp/hsqldb/cities;shutdown=true"]
         (def db {:classname "org.hsqldb.jdbcDriver"
           :subprotocol "hsqldb"
           :subname (str db-host ":" db-name)
           :user "SA"
           :password ""})

  (with-connection db
    (drop-table :cities)
    (create-table :cities
      [:id "varchar(10)" "PRIMARY KEY"]
      [:name :varchar]
      [:population :int]
      [:date_mod :date])
    (insert_proc)
      ))
(println "*** 終了 ***")
; -----------------------------------------------------------------
